test/services/NotProcessedFilesServiceSpec.scala

```diff
@@ -0,0 +1,230 @@
+        await(service.tryLock(expectingResult)) shouldBe Right(s"$jobName - JobAlreadyRunning")
+        capturedLogEvents.exists(_.getMessage == s"[$jobName] Locked because it might be running on another instance") shouldBe true
+        capturedLogEvents.exists(event => event.getLevel.levelStr == "INFO" && event.getMessage == s"[$jobName] Locked because it might be running on another instance") shouldBe true
+      }
+      verify(mockLockRepository, times(1)).takeLock(Matchers.eq(mongoLockId), Matchers.any(), Matchers.eq(releaseDuration))
+      verify(mockLockRepository, times(0)).releaseLock(Matchers.eq(mongoLockId), Matchers.any())
+    }
+
+    s"return $Left ${MongoLockResponses.UnknownException} if lock returns exception, release lock is still called and succeeds" in new Setup {
+      val expectingResult: Future[Right[Nothing, String]] = Future.successful(Right("hello"))
+      val exception = new Exception("woopsy")
+      when(mockLockRepository.takeLock(Matchers.eq(mongoLockId), Matchers.any(), Matchers.eq(releaseDuration)))
+        .thenReturn(Future.failed(exception))
+      when(mockLockRepository.releaseLock(Matchers.eq(mongoLockId), Matchers.any()))
+        .thenReturn(Future.successful(()))
+      withCaptureOfLoggingFrom(logger) { capturedLogEvents =>
+        await(service.tryLock(expectingResult)) shouldBe Left(MongoLockResponses.UnknownException(exception))
+        capturedLogEvents.exists(event => event.getLevel.levelStr == "INFO" && event.getMessage == s"[$jobName] Failed with exception") shouldBe true
+        eventually {
+          capturedLogEvents.exists(_.getMessage.contains(PagerDutyKeys.MONGO_LOCK_UNKNOWN_EXCEPTION))
+        }
+      }
+      verify(mockLockRepository, times(1)).takeLock(Matchers.eq(mongoLockId), Matchers.any(), Matchers.eq(releaseDuration))
+      verify(mockLockRepository, times(1)).releaseLock(Matchers.eq(mongoLockId), Matchers.any())
+    }
+
+    s"return $Left ${MongoLockResponses.UnknownException} if lock returns exception, release lock is still called and failed also" in new Setup {
+      val expectingResult: Future[Right[Nothing, String]] = Future.successful(Right("hello"))
+      val exception = new Exception("not again")
+      when(mockLockRepository.takeLock(Matchers.eq(mongoLockId), Matchers.any(), Matchers.eq(releaseDuration)))
+        .thenReturn(Future.failed(exception))
+      when(mockLockRepository.releaseLock(Matchers.eq(mongoLockId), Matchers.any()))
+        .thenReturn(Future.failed(exception))
+      withCaptureOfLoggingFrom(logger) { capturedLogEvents =>
+        await(service.tryLock(expectingResult)) shouldBe Left(MongoLockResponses.UnknownException(exception))
+        capturedLogEvents.exists(event => event.getLevel.levelStr == "INFO" && event.getMessage == s"[$jobName] Failed with exception") shouldBe true
+        eventually {
+          capturedLogEvents.exists(_.getMessage.contains(PagerDutyKeys.MONGO_LOCK_UNKNOWN_EXCEPTION))
```

Comment:
No assertion is made here - the tests will always pass. I can see this is the case in other files